### ISSUE TITLE
Make screenshots names time stamped instead of using 0-99

### ISSUE THREAD

**Issue by myself600** [diasurgical/devilutionX#4557]
**Describe**
The game does not create any screenshots past the 100 mark, due to 8.3 characters naming convention.

**To Reproduce**
1. Press [PrtScrn] key during gameplay (it will make you screen red, indicating that a screenshot is being created)
2. Repeat until you have all screen[00-99].PCX in %AppData%\diasurgical\devilution folder
3. Press [PrtScrn] key again during gameplay
4. Nothing happens

**Expected behavior**
Since 8.3 is no longer a relevant restriction the screenshots names should be time based: `Screenshot from YYYY-MM-DD hh-mm-ss.pcx` (for example: `Screenshot from 2022-04-24 07-17-37.pcx`) This naming allows them to be correctly sorted even when some are deleted. If a file with the name already exists append `-%d` to the end of the name (for example `Screenshot from 2022-04-24 07-17-37-1.pcx`), increase `%d` until no file with that name is encountered.

**Additional context**
`DevilutionX YYYY-MM-DD hh-mm-ss.pcx` is also an option for what to name the screen shots instead of the current name.

**Comment by DakkJaniels** [comment]
It doesn't make screenshots after 99 because that's a hardcoded limit, but looks simple enough to fix.

**Comment by qndel** [comment]
I'd say that if you need more than 100 screenshots, something is wrong with you :P

**Comment by DakkJaniels** [comment]
I would agree but it's a simple enough fix haha

**Comment by AJenbo** [comment]
It's less of a priority for me to work on this as the screenshot tool in Ubuntu takes over anyway, so I'm not personally affected, but I fully support the change :sweat_smile:

**Comment by DakkJaniels** [comment]
I already did something to just extend the number the screenshots (4.29B should be enough for anybody).  I can look into date stamps, but was having trouble finding a good way to do it.

**Comment by AJenbo** [comment]
I wonder if this works:
```cpp
const auto t = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
const auto creation_time = fmt::format("{:%FT%T%z}", fmt::localtime(t));
```

https://godbolt.org/z/K8s4Mc

**Comment by DakkJaniels** [comment]
I was doing this, but the string is ending up empty:
```cpp
	std::time_t now = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());

	std::string filename(30, '\0');
	std::strftime(&filename[0], filename.size(), "%Y-%m-%d %H:%M:%S", std::localtime(&now));

	SDL_Log(filename.c_str());
```

**Comment by DakkJaniels** [comment]
I got it - revised the PR accordingly.

**Comment by DakkJaniels** [comment]
I think it's more that:
1. Your OS might support screenshots in a way that doesn't require converting over PCX files (e.g., Windows Key + Shift + S)
2. You can always rename and/or move the files so that you have the ability to take more screenshots.

Regardless, I fixed your problem in #4558.

**Comment by AJenbo** [comment]
@DakkJaniels nice work. Now we just need to change it in to PNG files instead of PCX :)
